src/controllers/Document.ts

```diff
@@ -0,0 +1,805 @@
+import { BaseController } from './Base';
+import { SearchResult } from '../core/searchResult/SearchResultBase';
+import { DocumentsSearchResult } from '../core/searchResult/Document';
+import { JSONObject, Document, DocumentHit } from '../utils/interfaces';
+
+export class DocumentController extends BaseController {
+  constructor (kuzzle) {
+    super(kuzzle, 'document');
+  }
+
+  /**
+   * Counts documents in a collection.
+   *
+   * A query can be provided to alter the count result,
+   * otherwise returns the total number of documents in the collection.
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/document/count/
+   *
+   * @param index Index name
+   * @param collection Collection name
+   * @param query Query to match
+   * @param options Additional options
+   *    - `queuable` If true, queues the request during downtime, until connected to Kuzzle again
+   *
+   * @returns The number of matching documents
+   */
+  count (
+    index: string,
+    collection: string,
+    body: JSONObject,
```

Comment:
The body is optional:

```suggestion
    body: JSONObject = null,
```